lib/helper.coffee

```diff
@@ -0,0 +1,91 @@
+ChildProcess = require 'child_process'
+Path = require 'path'
+FS = require 'fs'
+{XRegExp} = require 'xregexp'
+
+class Helpers
+  @findFile: (directory, fileName) ->
+    climb = directory.split(Path.sep)
+    while climb.length
+      target = Path.join(climb.join(Path.sep), fileName)
+      if FS.existsSync(target)
+        return target
+      climb.pop()
+  @exec: (command, cwd, stdin) ->
+    executionTime = 5000
+    return new Promise (resolve, reject) ->
+      timeout = null
+      spawnedProcess = ChildProcess.exec "#{command.join(' ')}", {cwd}, (err, stdout, stderr)->
+        clearTimeout(timeout)
+        if err and (not stderr) and (not stdout)
+          reject(err)
+        else
+          resolve({stdout, stderr})
+      if stdin
+        spawnedProcess.stdin.write(stdin)
+        spawnedProcess.stdin.end()
+
+      timeout = setTimeout ->
+        spawnedProcess.kill()
+        errText = "command `#{command}` timed out after #{executionTime} ms"
+        atom.notifications.addError(
+          errText
+        )
+        reject(new Error(errText))
+      , executionTime
+  @createMessages:(textEditor, Regex, RegexFlags, Buffer) ->
+    Regex = XRegExp(Regex, RegexFlags)
+    textBuffer = textEditor.getBuffer()
+    filePath = textEditor.getPath()
+    toReturn = []
+    XRegExp.forEach Buffer, Regex, (match) ->
+      message = {}
+      if match.warning
+        message.type = 'Warning'
+      else if match.error or match.fail
+        message.type = 'Error'
```

Comment:
Something that immediately jumped at me. I see why you would use `match.warning` and `match.error` (don't see much use in `match.fail`, but sure, why not). But it should be possible to pass user types here as well, I think, e.g.
```cscript
else if match.type
    message.type = prefix+match.type
```
Not sure on how to pass prefix though.